src/libraries/System.Private.CoreLib/src/System/Globalization/OrdinalCasing.Icu.cs

```diff
@@ -427,6 +437,21 @@ internal static unsafe int LastIndexOf(ReadOnlySpan<char> source, ReadOnlySpan<c
             }
         }
 
+        private static void InitCasingTable()
+        {
+            ushort []?[] table = new ushort []?[s_casingTableInit.Length];
+            for (int i = 0; i < s_casingTableInit.Length; ++i)
+            {
+                if (s_casingTableInit[i] == 1)
+                    table[i] = s_noCasingPage;
+            }
+            table[0] = s_basicLatin;
+
+            // Publish
+            Interlocked.MemoryBarrier();
```

Comment:
You shouldn't need a `MemoryBarrier` or a `CompareExchange`. Publishing via `Volatile.Write` should be sufficient.